crypto/evp/evp_lib.c

```diff
@@ -32,14 +32,18 @@ int EVP_CIPHER_param_to_asn1(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
         case EVP_CIPH_CCM_MODE:
         case EVP_CIPH_XTS_MODE:
         case EVP_CIPH_OCB_MODE:
-            ret = -1;
+            ret = -2;
```

Comment:
I think its probably a bad idea to change the return code values for this function. We should keep it as -1. Perhaps set the reason code for EVPerr call below in a variable?

Reply:
I reverted the error return code to -1
and consequently reverted the line "A return value of -2 means that the cipher is not supported." in `doc/man3/EVP_EncryptInit.pod`.